### PR TITLE
32 add data aggregation density calculation script

### PR DESCRIPTION
**Pull request by mrdufour**
This was sloppy, but I got it done. Initially tried to create a function for the data aggregation and formatting code, but then abandoned that and just followed Jeremy's recipe. Which worked nicely. So, now when we run finish_setup() and third script (3_aggregate_format_data.R) appears in the 7_Annual_Summary folder.

### CODE REVIEW

File: inst/templates/aggregate_format_data.R [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+hacdat$STRATUM <- sapply(strsplit(hacdat$Region_name, "_"), '[',2)
+hacdat$GRID <- sapply(strsplit(hacdat$Region_name, "_"), '[',3)
+hacdat$LAYER <- sapply(strsplit(hacdat$Region_name, "_"), '[',4)
+
+## replace missing TS values with average TS from INTERVALS within same TRANSECT and LAYER
+hacdat$sigma_bs <- 10^(hacdat$TS_mean/10) # create sigma_bs
+hacdat$sigma_bs <- ifelse(hacdat$TS_mean == 9999, NA,  hacdat$sigma_bs) # assign NAs
+## impute missing sigma_bs_adj data - replace missing and Nv flagged sigma_bs values with mean of interval
+impute.mean <- function(x) replace(x, is.na(x), mean(x, na.rm = TRUE))
+hacdat <- plyr::ddply(hacdat, ~ STRATUM + GRID + LAYER, transform, sigma_bs = impute.mean(sigma_bs))
```
Comment (HoldenJe) [inline inst/templates/aggregate_format_data.R:60]: @mrdufour Line 60 in aggregate_format_data.R uses a plyr::ddply. Is there a way we can implement this without using plyr? Is there a dplyr alternative? Loading both plyr and dplyr causes namespace conflicts (maybe for select?) and it adds another package dependency.

Comment (HoldenJe) [reply inst/templates/aggregate_format_data.R:60]: @mrdufour can you test this code chunk in place of lines 59 and 60. I think it is doing the same thing as your code but I'm still getting NA values after I impute but I think it is a problem with my exports, or by problem - I might mean even with imputing I still have NA values and may actually need to impute at the STRATUM + LAYER to get a mean_sigma_bs.
```
hacdat %>% filter(is.na(sigma_bs))
sigma_bs_means <- hacdat %>% group_by(STRATUM, GRID, LAYER) %>% 
  filter(!is.na(sigma_bs)) %>% 
  summarize(mean_sigma_bs = mean(sigma_bs))

hacdat <- left_join(hacdat, sigma_bs_means, by=c('STRATUM', 'GRID', 'LAYER'))
hacdat <- hacdat %>% mutate(ifelse(is.na(sigma_bs), mean_sigma_bs, sigma_bs))
```

**Comment by HoldenJe** [comment]
981a6369ee83f7d39b59028f65d057e7bbbffbb4 fixes merge conflict with finish_setup

File: inst/templates/aggregate_format_data.R (hunk 1 above)
Comment (mrdufour) [reply inst/templates/aggregate_format_data.R:60]: @HoldenJe Yep, I'll test this afternoon. Also, I'll address the 'gather' function from 'plyr' as well.  Running out to finish up a trawl survey this morning.

**Comment by mrdufour** [comment]
@HoldenJe Check these out - both changes use 'dplyr'. Also, making use of 'Attribute' variable in histo create 'Bin_count' and 'Bin_TS_mean' columns

**Review by HoldenJe — APPROVED** [review]

**Comment by HoldenJe** [comment]
closes #37

**Comment by HoldenJe** [comment]
closes #37

**Pull request merged**